### PR TITLE
Implement addressbook

### PR DESCRIPTION
**Pull request by vic3king**
**What does this Merge request do?**

Adds endpoints for users to manage their contacts

**Description of Task to be completed?**

- create users as contacts on signup
- add and remove users from address book
- get users address book

**How should this be manually tested?**

xxx

**Any background context you want to provide?**

xxx

**What are the relevant Trello cards/stories?**

xxx

**Screenshots (if appropriate)**

xxx

**Questions:**

xxx

### CODE REVIEW

**Comment by vic3king** [comment]
@9trocode @otseobande @sojida Please review

File: server/middlewares/userContact.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  const {
+    params: { contactId },
+    user: { id: userId },
+  } = req;
+  const contact = await contactService.find({ id: contactId });
+
+  if (!contact) {
+    return res.status(404).json({
+      status: false,
+      error: 'Contact does not exist on the platform yet.',
```
Comment (9trocode) [inline server/middlewares/userContact.js:14]: @vic3king let's change this to user does not exist or what do you think cc @otseobande @sojida

File: server/middlewares/registration.js [hunk 2]
```diff
@@ -24,6 +26,15 @@ const registrationValidation = async (req, res, next) => {
       error: formattedMessage,
     });
   }
+
+  const user = await authService.find({ email });
+  if (user) {
+    return res.status(409).json({
+      status: false,
+      error: 'Please enter a unique email.',
```
Comment (9trocode) [inline server/middlewares/registration.js:34]: Let's format this to email already exist

File: server/controllers/userContactController.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+ */
+export default class userContactController {
+  /**
+   * @method add
+   * @description adds a new contact for a user
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} added contact
+   */
+  static async add(req, res) {
```
Comment (9trocode) [inline server/controllers/userContactController.js:17]: @vic3king can you please explain why you choose a static method over a class method ??

Comment (vic3king) [reply server/controllers/userContactController.js:17]: A static method can be called without having to instantiate the class.
https://medium.com/@yyang0903/static-objects-static-methods-in-es6-1c026dbb8bb1

**Comment by vic3king** [comment]
done @9trocode

**Comment by 9trocode** [comment]
> done @9trocode

@vic3king is this ready to get merge or I should hold on ??

**Comment by vic3king** [comment]
> > done @9trocode
> 
> @vic3king is this ready to get merge or I should hold on ??

@9trocode I've implemented the feedback you gave.

**Comment by 9trocode** [comment]
> > > done @9trocode
> > 
> > 
> > @vic3king is this ready to get merge or I should hold on ??
> 
> @9trocode I've implemented the feedback you gave.

Nice one @vic3king

**Pull request merged**